examples/azure_end_to_end/main.c

```diff
@@ -0,0 +1,284 @@
+
+#include "hw/azure_sphere_learning_path.h" // Hardware definition
+
+#include "dx_azure_iot.h"
+#include "dx_azure_iot.h"
+#include "dx_config.h"
+#include "dx_exit_codes.h"
+#include "dx_json_serializer.h"
+#include "dx_terminate.h"
+#include "dx_timer.h"
+#include "dx_utilities.h"
+#include <applibs/log.h>
+
+// https://docs.microsoft.com/en-us/azure/iot-pnp/overview-iot-plug-and-play
+#define IOT_PLUG_AND_PLAY_MODEL_ID "dtmi:com:example:azuresphere:labmonitor;1"
+#define NETWORK_INTERFACE "wlan0"
+
+// Forward declarations
+static void publish_message_handler(EventLoopTimer *eventLoopTimer);
+static void dt_desired_sample_rate_handler(DX_DEVICE_TWIN_BINDING *deviceTwinBinding);
+static void report_now_handler(EventLoopTimer *eventLoopTimer);
+static DX_DIRECT_METHOD_RESPONSE_CODE LightControlHandler(JSON_Value *json,
+                                                          DX_DIRECT_METHOD_BINDING *directMethodBinding,
+                                                          char **responseMsg);
+;
+
+DX_USER_CONFIG dx_config;
+
+/****************************************************************************************
+ * Telemetry message buffer property sets
+ ****************************************************************************************/
+
+// Number of bytes to allocate for the JSON telemetry message for IoT Hub/Central
+#define JSON_MESSAGE_BYTES 256
+static char msgBuffer[JSON_MESSAGE_BYTES] = {0};
+
+static DX_MESSAGE_PROPERTY *messageProperties[] = {&(DX_MESSAGE_PROPERTY){.key = "appid", .value = "hvac"},
+                                                   &(DX_MESSAGE_PROPERTY){.key = "type", .value = "telemetry"},
+                                                   &(DX_MESSAGE_PROPERTY){.key = "schema", .value = "1"}};
+
+static DX_MESSAGE_CONTENT_PROPERTIES contentProperties = {.contentEncoding = "utf-8",
+                                                          .contentType = "application/json"};
+
+static DX_GPIO_BINDING led = {
+    .pin = LED2, .name = "led", .direction = DX_OUTPUT, .initialState = GPIO_Value_Low, .invertPin = true};
+static DX_GPIO_BINDING network_led = {.pin = NETWORK_CONNECTED_LED,
+                                      .name = "network_led",
+                                      .direction = DX_OUTPUT,
+                                      .initialState = GPIO_Value_Low,
+                                      .invertPin = true};
+
+// All GPIOs added to gpio_set will be opened in InitPeripheralsAndHandlers
+DX_GPIO_BINDING *gpio_set[] = {&network_led, &led};
+
+/****************************************************************************************
+ * Timer Bindings
+ ****************************************************************************************/
+static DX_TIMER_BINDING publish_message = {
+    .period = {0, 250000000}, .name = "publish_message", .handler = publish_message_handler};
```

Comment:
instead of 25000000 we should start to use a MS multiplier constant so that it's easier to determine what the code is really asking for.

Reply:
fixed